co-log/src/Colog/Concurrent.hs

```diff
@@ -0,0 +1,129 @@
+-- |
+-- Sometimes an application may need to dump logs asynchonously.
+-- This may happen if the target where logs are dumped is
+-- slow and logs should be dumped in chunks, or if application
+-- should split CPU and I/O-bound tasks. In these cases it
+-- may make sense to run asynchronous logger.
+--
+-- Asynchronous logger runs in background thread, so users
+-- thread may dump logs asynchronously.
+--
+-- This module provides basic interface for the concurrent
+-- logging that may be either used as-is or can be used to
+-- build more complex concurrency abstractions.
+module Colog.Concurrent
+       ( -- $worker-thread
+         mkBackgroundThread
+       , runInBackgroundThread
+       -- $background-worker
+       , BackgroundWorker
+       , forkBackgroundLogger
+       , killBackgroundLogger
+       , convertToLogAction
+       ) where
+
+import Control.Concurrent
+import Control.Concurrent.STM.TBQueue
+
+import Colog.Core.Action (LogAction (..))
+import Colog.Concurrent.Internal
+
+-- $worker-thread
+-- Worker thread approach allows to create a single worker
+-- thread that will perform all logging. This approach allows
+-- user to use different loggers and change them on the flight.
+--
+-- With bracket resource control style code may look as
+-- following:
+--
+-- @
+-- bracket ('mkBackgroundThread' 4096)
+--         ('killBackgroundThread')
+--         $ \thread -> 
```

Comment:
Looks like `thread` variable in the documentation should be named as `worker` according to the code below